### PR TITLE
Circleci

### PR DESCRIPTION
**Pull request by ali2251**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -28,6 +28,7 @@
     "ganache-core": "^2.2.1",
     "request": "^2.88.0",
     "request-promise": "^4.2.2",
+    "truffle": "^4.1.14",
```
Comment (yahgwai) [inline package.json]: Lets add this to the state channel repo, then we can remove it from here.

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+      - save_cache:
+          key: dependency-cache-{{ checksum "package.json" }}
+          paths:
+            - ./node_modules
+      - run:
+          name: build kitsune
+          command: npm run build-kitsune
+      - run:
+          name: test
+          command: npm run test
```
Comment (yahgwai) [inline .circleci/config.yml]: This currently doesnt include the test-service npm task, we should add that here as well or add it to the 'test' npm task

File: package.json (hunk 1 above)
Comment (ali2251) [reply package.json]: sure

File: .circleci/config.yml (hunk 2 above)
Comment (ali2251) [reply .circleci/config.yml]: if the test isnt doing anything interesting, then we can keep only one, either `test` or `service-test`

File: .circleci/config.yml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+      - run:
+          name: Deploy Over SSH
+          command: |
+            ssh $SSH_USER@$SSH_HOST "ls; cd mvp-api; git pull; node -v; yarn install; yarn build; yarn start"
+workflows:
+  version: 2
+  build-and-deploy:
+    jobs:
+      - build
+      - deploy:
```
[the remark this answers is not in the document]
Comment (ali2251) [reply .circleci/config.yml]: yup, will remove it

File: .circleci/config.yml (hunk 2 above)
Comment (yahgwai) [reply .circleci/config.yml]: both are useful atm. We should run the functionality of both ```test``` and ```service-test```, either by combining them both into a command - or by just running both commands

Comment (ali2251) [reply .circleci/config.yml]: added both

File: .circleci/config.yml [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+          key: dependency-cache-{{ checksum "package.json" }}
+          paths:
+            - ./node_modules
+      - run:
+          name: build kitsune
+          command: npm run build-kitsune
+      - run:
+          name: test
+          command: npm run test
+  deploy:
```
[the remark this answers is not in the document]
Comment (ali2251) [reply .circleci/config.yml]: we have both test and service test now

File: .circleci/config.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (ali2251) [reply .circleci/config.yml]: done

**Comment by ali2251** [comment]
Let me know if there is anything else

**Pull request merged**